### PR TITLE
docs: :memo: Adding documentation for the support of lon in the latLng function, resolves 8509

### PR DESCRIPTION
**Pull request by brianferry**
### Issue

#8509 

### What I Did

Updated the reference docs and the comments in the LatLng.js file to include notes outlining the ability to use `lon` instead of `lng` when passing in an object to a function that uses the LatLng object.  References to where this functionality came from can be found in the issue and are present for compatibility with other services such as D3.js.

### CODE REVIEW

**Review by jonkoops — CHANGES_REQUESTED** [review]

File: src/geo/LatLng.js [hunk 1]
```diff
@@ -17,7 +17,7 @@ import {toLatLngBounds} from './LatLngBounds';
  *
  * ```
  * map.panTo([50, 30]);
- * map.panTo({lat: 50, lng: 30});
+ * map.panTo({lat: 50, lng: 30}); // Note: `lon` can be used interchangably with `lng` in objects.
```
Comment (mourner) [inline src/geo/LatLng.js:20]: I think it's clearer with an example:

```suggestion
 * map.panTo({lat: 50, lng: 30});
 * map.panTo({lat: 50, lon: 30});
```

File: src/geo/LatLng.js [hunk 2]
```diff
@@ -109,6 +109,7 @@ LatLng.prototype = {
 // @alternative
 // @factory L.latLng(coords: Object): LatLng
 // Expects an plain object of the form `{lat: Number, lng: Number}` or `{lat: Number, lng: Number, alt: Number}` instead.
+// As noted previously, `lon` can be used in place of `lng` when passing in an object to this function.
```
[the remark this answers is not in the document]
Comment (Falke-Design) [reply src/geo/LatLng.js:112]: I have no big opinion about this but if the users passes `{ lat: 1, lon: 3}` the LatLng object will only accessible over `latlng.lng`. So it is more a compatibility feature then `lng`

[the remark this answers is not in the document]
Comment (brianferry) [reply src/geo/LatLng.js]: These are all fair points, while they're both valid means of setting the values in this function there's only one getter and that's the .lng value.  What do we think about outlining that like "You can also use `lon` in place of `lng` in the object form.  Both `lon` and `lng` are assigned to the `.lng` property."?

[the remark this answers is not in the document]
Comment (Falke-Design) [reply src/geo/LatLng.js]: Sounds good to me 👍 @mourner, @jonkoops?

File: docs/reference.html [hunk 3]
```diff
@@ -19248,7 +19248,7 @@ <h3 id='latlng-example'>Usage example</h3>
 <p>All Leaflet methods that accept LatLng objects also accept them in a simple Array form and simple object form (unless noted otherwise), so these lines are equivalent:</p>
 <pre><code>map.panTo([50, 30]);
 map.panTo({lng: 30, lat: 50});
-map.panTo({lat: 50, lng: 30});
+map.panTo({lat: 50, lng: 30}); // Note: `lon` can be used interchangably with `lng` in objects.
```
Comment (Falke-Design) [inline docs/reference.html:19251]: Please revert this file, it is generated over a script. It's enough to change the doc in `LatLng.js`

**Review by Falke-Design — CHANGES_REQUESTED** [review]

**Review by jonkoops — APPROVED** [review]
Changes look good. As @Falke-Design mentioned we don't need to update `docs/reference.html`, so please revert that file.

File: src/geo/LatLng.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (jonkoops) [reply src/geo/LatLng.js]: I'm good with how the changes are now. I agree with @mourner not to be too verbose.

**Comment by brianferry** [comment]
Thanks @jonkoops, I have reverted that file, thank you!

**Review by Falke-Design — APPROVED** [review]

**Pull request merged**